### ISSUE TITLE
non-returning Generator should infer `Never` for 3rd type parameter

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#6707]
**Describe the bug**

unlike normal function, if generator function always raises Exception, third type parameter of `Generator` is inferred as `None` instead of `Never`.

**Code or Screenshots**

```python
from typing import Generator
from typing_extensions import Never

class P:
    def g(self) -> Generator[int, Never, int]:
        ...

class D(P):
    def g(self):
        yield 3
        raise RuntimeError()

reveal_type(D().g)
```
```console
$ pyright gt.py 
gt.py
  gt.py:9:9 - error: Method "g" overrides class "P" in an incompatible manner
    Return type mismatch: base method returns type "Generator[int, Never, int]", override returns type "Generator[Literal[3], Any, None]"
      "Generator[Literal[3], Any, None]" is incompatible with "Generator[int, Never, int]"
        Type parameter "_ReturnT_co@Generator" is covariant, but "None" is not a subtype of "int"
          "None" is incompatible with "int" (reportIncompatibleMethodOverride)
  gt.py:13:13 - information: Type of "D().g" is "() -> Generator[Literal[3], Any, None]"
1 error, 0 warnings, 1 information 
```

**VS Code extension or command-line**

command-line tool, 1.1.339

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.
